Detectors/EMCAL/calibration/src/EMCALChannelCalibrator.cxx

```diff
@@ -0,0 +1,123 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "EMCALCalibration/EMCALChannelCalibrator.h"
+#include "Framework/Logger.h"
+#include "CommonUtils/MemFileHelper.h"
+#include "CCDB/CcdbApi.h"
+#include "DetectorsCalibration/Utils.h"
+#include <boost/histogram.hpp>
+#include <boost/histogram/ostream.hpp>
+#include <boost/format.hpp>
+#include <cassert>
+#include <iostream>
+#include <sstream>
+#include <TStopwatch.h>
+
+namespace o2
+{
+namespace emcal
+{
+
+using Slot = o2::calibration::TimeSlot<o2::emcal::EMCALChannelData>;
+using clbUtils = o2::calibration::Utils;
+using boost::histogram::indexed;
+
+//===================================================================
+//_____________________________________________
+void EMCALChannelData::fill(const gsl::span<const o2::emcal::Cell> data)
+{
+  // fill container
+  for (auto cell : data) {
+    Double_t cellEnergy = cell.getEnergy();
+    Int_t id = cell.getTower();
+    LOG(DEBUG) << "inserting in cell ID " << id << ": energy = " << cellEnergy;
+    mHisto(cellEnergy, id);
+  }
+}
+//_____________________________________________
+void EMCALChannelData::print()
+{
+  // print a useful message
+  LOG(DEBUG) << "printing a useful message about the histogram";
+}
```

Comment:
Please implement operator<< for EMCALChannelData, this can then be reused in any kinds of streams. The implementation if print would then simply reduce to
``` c++
void EMCALChannelData::print()
{
  // print a useful message
  LOG(DEBUG) << *this;
}
```